osbuild/util/jsoncomm.py

```diff
@@ -364,13 +388,30 @@ def recv(self):
             raise BufferError
 
         try:
-            payload = json.loads(msg[0])
+            payload = json.loads(serialized)
         except json.JSONDecodeError as e:
             raise BufferError from e
 
         return (payload, fdset, msg[3])
 
-    def send(self, payload: object, *, fds: Optional[list] = None):
+    def _sendmsg_handle_emsgsize(self, serialized, fds):
+        cmsg = []
+        if fds:
+            cmsg.append((socket.SOL_SOCKET, socket.SCM_RIGHTS, array.array("i", fds)))
+        try:
+            self._socket.sendmsg([serialized], cmsg, 0)
+        except OSError as exc:
+            if exc.errno == errno.EMSGSIZE:
+                with memfd("jsoncomm/payload") as fd_payload:
+                    os.write(fd_payload, serialized)
+                    os.lseek(fd_payload, 0, 0)
+                    cmsg = []
+                    cmsg.append((socket.SOL_SOCKET, socket.SCM_RIGHTS, array.array("i", [fd_payload] + fds)))
+                    self._socket.sendmsg([ARGS_VIA_FD_MARKER], cmsg, 0)
+                    return
```

Comment:
Not sure.  This feels a tiny bit too automatic.  Is it difficult to know whether we'll exceed the msgsize ahead of time instead?  Even if it's a very conservative estimate that favours using fds more often than required, I think it would be worth it for the clarity.

Reply:
In general I feel any time we're using file descriptors to get around large message sizes we're doing things that we shouldn't be doing. It makes sense from a privilege perspective that we want to pass fds into sandboxed processes.

Anyhow, that's not what this PR is here to change. I think the easier-to-ask-for-forgiveness approach here is an OK way of doing this.

Reply:
I tweaked the code now to read `/proc/sys/net/core/wmem_max` insead of blindly sending, it's a TOCTOU but that does not matter here. I have no strong opinions either way. Fwiw, when looking at this I was wondering if we should move to a real streaming socket and streaming json protocol (like json-seq) but that is a larger change of course.